### DISCUSSION TITLE
Stop the query when removing a query from the log

### DISCUSSION THREAD

**Discussion by giftig** [apache/superset#24866] [Ideas]
I think this one could almost considered a bug but I'm raising it here instead of as an issue since it's borderline.

While working on https://github.com/apache/superset/issues/24858 I toyed with edge cases around refreshing the page / navigating away (with the "cancel query on page unload" database switch off) and saw that the query continues to execute and we continue to poll the results, and the results will eventually be available in the history tab as a result. That's fine.

When you delete the running entry from history, however, we stop polling for the results but the query continues to execute pointlessly; presumably when it finishes it'll get an error because it's saving to a `Query` that we've deleted. There's no other way in history to stop the query execution, either.

So it seems to me that deleting a `Query` like this should also trigger the query to be cancelled if it's still running. This seems like a fairly easy change if it's simply missing, so I can raise a PR if there's consensus that that's how it should work.

**Comment by rusackas** [comment]
That seems like a valid improvement and fairly non-controversial to me. I'm trying to think of edge cases that might be dangerous, but nothing of real concern springs to mind. Pinging @yousoph @eschutho @villebro @john-bodley in case any of them have opinions they'd like to share. Personally, I'd say "go for it" with the PR if it's easy - that provides a different venue to have this discussion, if that helps the cause :)

**Reply by giftig** [reply]
Ah thanks for weighing in, Evan; I'd forgotten about this one amongst the other changes I'm working on but I should be able to make a PR for this at some point. I'll focus on the PRs I already have open first though, I think, I don't want to bombard everyone with PRs too much.
